dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM python:3.7-slim-buster as base
+
+# Perform common operations, dependency installation etc...
+RUN apt-get update && apt-get install -y curl
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+ENV PATH = "${PATH}:/root/.poetry/bin"
+
+RUN mkdir TO-DO-APP
+COPY . ./TO-DO-APP
+
+WORKDIR /TO-DO-APP
+RUN poetry install
```

Comment:
You can improve the speed of rebuilds if you copy the poetry files, run poetry install, and then copy the rest of the files. That way it won't need to redo the poetry install command whenever files are changed.

The COPY command could be moved into the production stage if the intention is to run the development image with a mounted volume.

The `poetry install` command could be moved into the production/development stages so that e.g. you can run `poetry install --no-dev` for the production image.

Reply:
@hugh-emerson What you mean by Poetry files? Should I `cp poetry.lock ` , `RUN poetry install` and then copy rest of files you mean?

Reply:
@hugh-emerson What you mean by Poetry files? Should I `cp poetry.lock ` , `RUN poetry install` and then copy rest of files you mean?

Reply:
You need poetry.lock and pyproject.toml. And you mean `COPY` not `cp`.

E.g. `COPY poetry.lock pyproject.toml .` (if you've set WORKDIR already). And then yes, run poetry install and copy everything.